src/mod_muc.erl

```diff
@@ -752,13 +726,25 @@ start_new_room(Host, ServerHost, Access, Room,
                                RoomShaper, HttpAuthPool, Opts)
     end.
 
+handle_room_registration(Host, Room, Pid) ->
+    case register_room(Host, Room, Pid) of
+        {_, ok} ->
+            {ok, Pid};
+        {_, {exists, OldPid}} ->
+            exit(Pid, normal),
+            {exists, OldPid}
```

Comment:
The function will stop the newly started process associated with Pid, but the calling function will still need OldPid in order to route the message to the room, will it not? Would the message not be lost otherwise?

Reply:
Yes, but `{ok, OldPid}` may be returned instead.